pr47-core/src/vm/al31f/executor/async_executor.rs

```diff
@@ -40,6 +42,40 @@ pub async fn create_vm_main_thread<A: Alloc>(
     ret
 }
 
+#[allow(unused)]
+unsafe fn exception_unwind_stack<A: Alloc>(
+    program: &CompiledProgram<A>,
+    exception: &Exception,
+    stack: &mut Stack,
+    insc_ptr: usize
+) -> Option<(StackSlice, usize)> {
+    let checked_exception: &CheckedException =
+        if let Exception::CheckedException(ce /*: &CheckedException*/) = exception {
+            ce
+        } else {
+            return None;
+        };
+
+    let mut insc_ptr: usize = insc_ptr;
+    let stack_frame: &FrameInfo = stack.frames.last().unchecked_unwrap();
+    let func_id: usize = stack_frame.func_id;
+    let compiled_function: &CompiledFunction = &program.functions[func_id];
+    if let Some(exc_handlers /*: &Box<[ExceptionHandlingBlock]>*/)
+        = &compiled_function.exc_handlers
+    {
+        for exc_handler /*: &ExceptionHandlingBlock*/ in exc_handlers.as_ref().iter() {
+            let (start_insc, end_insc): (usize, usize) = exc_handler.insc_ptr_range;
+            if insc_ptr >= start_insc
+                && insc_ptr <= end_insc
+                && /* exc_handler.exception_id == checked_exception.type_id() */ false {
```

Comment:
By this time [`std::error::Error::type_id`](https://github.com/rust-lang/rust/issues/60784) is still unstable. We need some other way to achieve this stuff.

Reply:
**[MID]** Perhaps we can make a `throw!` macro for creating a checked exception. Or maybe a macro is not necessary, just using functions will be fine?